operator/internal/central/extensions/reconcile_pvc.go

```diff
@@ -119,32 +63,67 @@ func convertDBPersistenceToPersistence(p *platform.DBPersistence, target PVCTarg
 //     database backups
 //
 // A nil return value indicates that no persistent volume should be provisioned for the respective target.
-func getPersistenceByTarget(central *platform.Central, target PVCTarget, log logr.Logger) (*platform.Persistence, error) {
+func getPersistenceByTarget(central *platform.CentralComponentSpec, target PVCTarget) (*platform.DBPersistence, error) {
 	switch target {
 	case PVCTargetCentral:
 		return nil, nil
 	case PVCTargetCentralDB, PVCTargetCentralDBBackup:
-		if !central.Spec.Central.ShouldManageDB() {
+		if !central.ShouldManageDB() {
 			return nil, nil
 		}
-		dbPersistence := central.Spec.Central.GetDB().GetPersistence()
-		if dbPersistence == nil {
-			dbPersistence = &platform.DBPersistence{}
+		p := central.GetDB().GetPersistence()
+		if p == nil {
+			p = &platform.DBPersistence{} // important semantically, see func description
 		}
 
-		return convertDBPersistenceToPersistence(dbPersistence, target)
+		if p.HostPath != nil {
+			return &platform.DBPersistence{
+				HostPath: p.HostPath,
+			}, nil
+		}
+		if target == PVCTargetCentralDBBackup {
+			return getBackupDBPersistence(p.GetPersistentVolumeClaim().DeepCopy())
+		}
+		return p.DeepCopy(), nil
 	default:
 		return nil, errors.Errorf("unknown pvc target %q", target)
 	}
 }
 
+// getBackupDBPersistence returns the DBPersistence for the _backup_ central-db volume, based on a copy of the
+// _main_ central-db volume config. It is free to trash or reuse the provided pvc object.
+func getBackupDBPersistence(pvc *platform.DBPersistentVolumeClaim) (*platform.DBPersistence, error) {
+	if pvc != nil && pvc.ClaimName != nil {
+		// If a ClaimName is specified, derive the backup PVC ClamName from it.
+		backupName := common.GetBackupClaimName(*pvc.ClaimName)
+		pvc.ClaimName = &backupName
+	}
+
+	if pvc != nil && pvc.Size != nil {
+		// If a Size is specified, derive the backup PVC Size from it as
+		// well, the rule of thumb is that it should be twice as large, to
+		// accommodate the backup and one restore copy.
+		quantity, err := resource.ParseQuantity(*pvc.Size)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to calculate backup volume size")
```

Comment:
I know this is just moved code, but maybe it would be more accurate at this point to say something like

```suggestion
			return nil, errors.Wrapf(err, "failed to parse backup volume size %q", *pvc.Size)
```
since this is just an error at this point caused by parsing.

Reply:
Well, we are parsing the non-backup PVC size. I'm changing it so that now the message will read:
```
failed to calculate backup volume size: parsing main central-db PVC size "1 Badger" failed: quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'
```

Reply:
Also adding a test case.